app/code/Tasks/FeedbackManager/Controller/Manage/Save.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Tasks\FeedbackManager\Controller\Manage;
+
+use Magento\Customer\Controller\AbstractAccount;
+use Magento\Framework\App\Action\Context;
+use Magento\Customer\Model\Session;
+
+class Save extends \Magento\Framework\App\Action\Action
+{
+
+    protected $feedbackFactory;
+    protected $customerSession;
+    protected $messageManager;
+
+    public function __construct(Context $context, \Tasks\FeedbackManager\Model\FeedbackFactory $feedbackFactory, Session $customerSession, \Magento\Framework\Message\ManagerInterface $messageManager)
+    {
+        $this->feedbackFactory = $feedbackFactory;
+        $this->customerSession = $customerSession;
+        $this->messageManager = $messageManager;
+        parent::__construct($context);
+    }
+
+    public function execute(){
+        $data = $this->getRequest()->getParams();
+        $customerId = $this->customerSession->getCustomerId();
+        if (isset($data['id']) && $data['id']) { // Editing the form. Only in Edit form, we have 'id' set
+            $isAuthorised = $this->feedbackFactory->create()
+                                        ->getCollection()
+                                        ->addFieldToFilter('user_id', $customerId)
+                                        ->addFieldToFilter('entity_id', $data['id'])
+                                        ->getSize();
+            if (!$isAuthorised) {
+                $this->messageManager->addError(__('You are not authorised to edit this blog.'));
+                return $this->resultRedirectFactory->create()->setPath('feedback/manage');
+            } else {
+                $model = $this->feedbackFactory->create()->load($data['id']); // select * from feedbackmanager_feedback where entity_id = 4
+                // Updating the table based on the user inputs
+                $model->setCustomerFirstname($data['customer_firstname'])
+                        ->setCustomerLastname($data['customer_lastname'])
+                        ->setCustomerEmail($data['customer_email'])
+                    ->setComment($data['comment'])
+                    ->save();
+                $this->messageManager->addSuccess(__('You have updated the feedback successfully.'));
+            }
+        } else {    // Adding the feedback.
```

Comment:
@gayathri-aravind Too much else. Avoud else as much as possible and use guard clause

Reply:
Used guard clause. Commit: 97176bc9f722f2f98d8297092de9d998c8c3d170